pkg/events/queue/queue_mem_list.go

```diff
@@ -0,0 +1,118 @@
+package queue
+
+import (
+	"container/list"
+	"fmt"
+	"sync"
+
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+type eventQueueMem struct {
+	mutex                *sync.Mutex
+	cond                 *sync.Cond
+	cache                *list.List
+	maxAmountOfEvents    int // max number of cached events possible
+	eventsCacheMemSizeMB int
+	verbose              string
+}
+
+func NewEventQueueMem(queueSizeMb int) CacheConfig {
+	q := &eventQueueMem{
+		eventsCacheMemSizeMB: queueSizeMb,
+	}
+	q.setup()
+	q.verbose = fmt.Sprintf("In-Memory Event Queue (Size = %d MB)", q.eventsCacheMemSizeMB)
+	return q
+}
+
+func (q *eventQueueMem) String() string {
+	return q.verbose
+}
+
+func (q *eventQueueMem) setup() {
+	q.mutex = new(sync.Mutex)
+	q.cond = sync.NewCond(q.mutex)
+
+	// set queue size and init queue
+	q.maxAmountOfEvents = q.getQueueSizeInEvents()
+	q.cache = new(list.List)
+}
+
+// Enqueue pushes an event into the queue (may block until queue is available)
+func (q *eventQueueMem) Enqueue(evt *trace.Event) {
+	q.cond.L.Lock()
+	// enqueue waits for de-queuing if cache is full (using >= instead of == to be in the safe side...)
+	for q.cache.Len() >= q.maxAmountOfEvents {
+		q.cond.Wait()
+	}
+
+	q.cache.PushBack(*evt)
+	q.cond.L.Unlock()
+	q.cond.Signal() // unblock dequeue if needed
+
+	evt = nil
+}
+
+// Dequeue pops an event from the queue
+func (q *eventQueueMem) Dequeue() *trace.Event {
+	q.cond.L.Lock()
+
+	// dequeue waits for en-queueing if cache is empty
+	for q.cache.Len() == 0 {
+		q.cond.Wait()
+	}
+
+	e := q.cache.Front()
+	event, ok := e.Value.(trace.Event)
+	if !ok {
+		q.cond.L.Unlock()
+		return nil
+	}
+	q.cache.Remove(e)
+	q.cond.L.Unlock()
+	q.cond.Signal() // unblock enqueue if needed
+
+	return &event
+}
+
+// getQueueSizeInEvents returns size of the fifo queue, in # of events, based on
+// the host size
+func (q *eventQueueMem) getQueueSizeInEvents() int {
+	// eventSize is the memory footprint per event, in bytes. This is NOT the
+	// size of a single event, but the overall impact in memory consumption to
+	// each cached event (defined by experimentation)
+	eventSize := 1024
+
+	KBtoB := func(amountInKB int) int {
+		return amountInKB * 1024
+	}
+	MBtoKB := func(amountInMB int) int {
+		return amountInMB * 1024
+	}
+	GBtoMB := func(amountInGB int) int {
+		return amountInGB * 1024
+	}
+	AmountOfEvents := func(amountInMB int) int {
+		return MBtoKB(KBtoB(amountInMB)) / eventSize
```

Comment:
@rafaeldtinoco FYI shouldn't this be`KbtoB(MBtoKB(amountInMB))?`

Reply:
btw, asking you as was part of your commits.

Reply:
could be if eventSize was 1 and not 1024 (if I got what you mean correctly). 

If I did that, then eventSize would always have to be 1024 multiple (1k, 2k, 4k). The way it is, number can be 768 or 1280 (which makes a big difference at the end).

The most correct thing to do here would be to calculate the eventSize based on an existing number (trace.Event) BUT I'm afraid that it is more complex than just that (considering garbage collection and many other overhead from runtime). 

I measured different cache sizes and found out 1024 was something close to a constant for calculating cache sizes VS amount of events (this won't be true if we start having other event types in the future).

Reply:
what I am saying is that your are using function `KBtoB(amountInMB)` and accepting something which is in units of MB. I don't see any constellation of input where this could be correct. I think should simply be the opposite: 
`KbtoB(MBtoKB(amountInMB))`